py/malloc.c

```diff
@@ -53,11 +54,19 @@
 // freely accessed - for interfacing with system and 3rd-party libs for
 // example. On the other hand, some (e.g. bare-metal) ports may use GC
 // heap as system heap, so, to avoid warnings, we do undef's first.
+// CIRCUITPY-CHANGE: Add selective collect support to malloc to optimize GC for large buffers
 #undef malloc
 #undef free
 #undef realloc
-#define malloc(b) gc_alloc((b), false)
-#define malloc_with_finaliser(b) gc_alloc((b), true)
+#if MICROPY_ENABLE_SELECTIVE_COLLECT
+#define malloc(b) gc_alloc((b), GC_ALLOC_FLAG_DO_NOT_COLLECT)
+#define malloc_with_collect(b) gc_alloc((b), 0)
+#define malloc_without_collect(b) gc_alloc((b), GC_ALLOC_FLAG_DO_NOT_COLLECT)
```

Comment:
I wonder if it would be safer to do one of the following:
1. Remove `malloc()` altogether and require `malloc_with_collect()` or `malloc_without_collect()`. Then it's a conscious decision each time.
2. Or, make `malloc()` collect by default, so any `malloc()`s that were missed are collected, which is safer.